File: c/collatz-conjecture/src/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
  if (start <= 0)
  {
    return -1;
  }
  int steps = 0;
  while (start > 1)
  {
    if (start % 2 == 0)
    {
      start /= 2;
    }
    else
    {
      start = 3 * start + 1;
    }
    steps++;
  }
  return steps;
}

File: kotlin/collatz-conjecture/src/main/kotlin/CollatzConjecture.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {
        if (start < 1) {
            throw IllegalArgumentException("Only natural numbers are allowed")
        }
        var steps = 0
        var n = start
        while (n > 1) {
            if (n % 2 == 0) {
                n /= 2
            } else {
                n = 3 * n + 1
            }
            steps += 1
        }
        return steps
    }
}
